logistics_project/apps/malawi/warehouse/report_views/stock_status.py

```diff
@@ -72,70 +73,97 @@ def get_stock_status_by_product_table(self, reporting_supply_point, is_facility)
+        )
+
+        if is_facility:
+            supply_points = hsa_supply_points_below(reporting_supply_point.location)
+        else:
+            supply_points = facility_supply_points_below(reporting_supply_point.location)
+
+        for supply_point in supply_points:
+            temp = [supply_point.name]
+            for product in products:
+                ps = ProductStock.objects.filter(supply_point=supply_point, product=product)
+                if ps.count():
+                    mr = ps[0].months_remaining
+                    if mr:
+                        temp.append('%.1f' % mr)
                     else:
                         temp.append('-')
-                hsa_table["data"].append(temp)
+                else:
+                    temp.append('-')
+            table["data"].append(temp)
+
+        return table
 
-            location_table = {} # don't show the location table for HSA's only
+    def get_stock_status_across_location_table(self, request, reporting_supply_point, selected_product, is_facility):
+        table = {
+            "id": "location-table",
+            "is_datatable": True,
+            "is_downloadable": True,
+        }
+        reporter = 'Facility' if is_facility else 'HSA'
+        headings = [(header % reporter) for header in [
+            "%% %s Stocked Out",
+            "%% %s Under",
+            "%% %s Adequate",
+            "%% %s Overstocked",
+            "%% %s Not Reported",
+        ]]
+
+        if is_district(reporting_supply_point):
+            table["location_type"] = "Facility"
+            table["header"] = [table["location_type"]] + headings
+            table["data"] = self._get_product_status_table(
+                facility_supply_points_below(reporting_supply_point.location).order_by('name'),
+                selected_product,
+                current_report_period(),
+            )
         elif is_country(reporting_supply_point):
-            location_table["location_type"] = "District"
-            location_table["header"] = [location_table["location_type"]] + headings
-            location_table["data"] = self._get_product_status_table(
+            table["location_type"] = "District"
+            table["header"] = [table["location_type"]] + headings
+            table["data"] = self._get_product_status_table(
                 get_district_supply_points(request.user.is_superuser).order_by('name'),
                 selected_product,
                 current_report_period(),
             )
 
-            
+        return table
+
+    def custom_context(self, request):
+        selected_type = self.get_selected_product_type(request)
+        selected_product = self.get_selected_product(request)
+        reporting_supply_point = self.get_reporting_supply_point(request)
+
+        months_of_stock_table = None
+        stock_status_across_location_table = None
+
+        if (
+            (is_facility(reporting_supply_point) and not request.is_facility) or
```

Comment:
agree, see my note above about renaming the `is_facility` flag globally. also agree with extracting that logic into its own method.

i also think this becomes much clearer if we take the approach of using `base_level` as the data point, which takes one of a list of possible values:

    (is_facility(reporting_supply_point) and request.base_level == 'h') or
    (is_district(reporting_supply_point) and request.base_level == 'f')